### PR TITLE
SUIT-9842 Check if Suitest JS API is running the latest version on launcher start-up.

### PR DESCRIPTION
**Pull request by JeKachan**
* displayed warning in child process if launcher version different from current suitest library.
 * displayed warning after finishing interactive/automated session if launcher has not latest available version.
 * added packageMetadataHelper for working with package meta data.
 * removed unused environment constant

### CODE REVIEW

File: lib/testLauncher/SuitestLauncher.js [hunk 1]
```diff
@@ -168,6 +175,7 @@ class SuitestLauncher {
 							[envVars.SUITEST_CONFIG_LOG_LEVEL]: config.logLevel,
 							[envVars.SUITEST_CONFIG_DISALLOW_CRASH_REPORTS]: config.disallowCrashReports,
 							[envVars.SUITEST_CONFIG_CONTINUE_ON_FATAL_ERROR]: config.continueOnFatalError,
+							[envVars.SUITEST_LAUNCHER_VERSION]: '1.4.0' || version,
```
Comment (givehug) [inline lib/testLauncher/SuitestLauncher.js]: what is the purpose of hardcoding '1.4.0' ?

File: lib/testLauncher/SuitestLauncher.js [hunk 2]
```diff
@@ -27,12 +27,16 @@ const {captureException} = require('../utils/sentry/Raven');
...
+const {version} = require('../../package.json');
+const {fetchLatestSuitestVersion, warnNewVersionAvailable} = require('../utils/packageMetadataHelper');
 
 class SuitestLauncher {
 	constructor(ownArgv = {}, restArgs = []) {
 		this.ownArgv = ownArgv;
 		this.restArgs = restArgs;
+		this.latestSuitestVersion = '';
 
+		fetchLatestSuitestVersion().then(version => this.latestSuitestVersion = version);
```
Comment (givehug) [inline lib/testLauncher/SuitestLauncher.js]: is it possible that request resolves later than suitest launches child processes?

File: lib/texts.js [hunk 3]
```diff
@@ -132,6 +132,8 @@ ${leaves}`,
 	'tl.startLogRecording': (device) => `Logs for device: ${device} will be written to`,
 	'tl.logDirDescription': () => 'If defined, additional logging to specified folder will be performed.',
 	'tl.inspectOnlyForInteractiveMode': () => 'Please use interactive mode to enable debugging.',
+	'tl.differentLauncherAndLibVersions': template`The version of suitest library (${0}) provided from your script is different from suitest launcher (${1})`,
+	'tl.newVersionAvailable': template`The new version of suitest library are available (${0}).`,
```
Comment (givehug) [inline lib/texts.js]: `library is available`

File: lib/utils/packageMetadataHelper.js [hunk 4]
```diff
@@ -0,0 +1,58 @@
+const semver = require('semver');
+const packageJson = require('package-json');
+const envVars = require('../constants/enviroment');
+const texts = require('../texts');
+const logger = require('./logger');
+const {version} = require('../../package.json');
+
+const LIB_NAME = 'suitest-js-api';
```
Comment (givehug) [inline lib/utils/packageMetadataHelper.js]: can be used in constants/index.js

Comment (AntonSBilan) [inline lib/utils/packageMetadataHelper.js]: @JeKachan maybe we can get name and version just from package.json?

**Comment by andriichumak** [comment]
@JeKachan @givehug by the way, this is a good example of out UTs being unstable. I'm pretty sure that if I run it again it will pass. Could someone have a look into it, please?

File: package.json [hunk 5]
```diff
@@ -27,7 +27,7 @@
     "suitest": "bin/suitest"
   },
   "scripts": {
-    "test": "mocha --no-timeouts",
+    "test": "mocha",
```
Comment (andriichumak) [inline package.json]: Was it the reason for UTs being unstable? How come?

Comment (JeKachan) [reply package.json]: there was for developming only. I added it when I tried make success mock for package-json library http request

**Pull request merged**